src/clarity-angular/popover/dropdown/dropdown.ts

```diff
@@ -15,10 +18,25 @@ import {IfOpenService} from "../../utils/conditional/if-open.service";
         // FIXME: remove this as soon as we stop supporting this old <div class="dropdown-menu"> syntax
         "[class.open]": "ifOpenService.open"
     },
-    providers: [IfOpenService]
+    providers: [
+        IfOpenService, {
+            provide: DropdownService,
+            useFactory: clrDropdownProviderFactory,
+            deps: [[new Optional(), new SkipSelf(), DropdownService]]
+        }
+    ]
 })
-export class Dropdown {
-    constructor(@SkipSelf() @Optional() public parent: Dropdown, public ifOpenService: IfOpenService) {}
+export class Dropdown implements OnDestroy {
+    private _subscription: Subscription;
+
+    constructor(@SkipSelf() @Optional() public parent: Dropdown, public ifOpenService: IfOpenService,
+                public dropdownService: DropdownService) {
```

Comment:
This doesn't need to be a property on the object at the moment, you're only using it in the constructor. The linter will fail once we add typing. So you can just remove the `public` keyword here.